src/main/java/com/codejoust/main/service/ProblemService.java

```diff
@@ -401,15 +416,19 @@ private void validateGsonParseable(String input, ProblemIOType type) {
         return problemDtos;
     }
 
-    public List<ProblemTagDto> getAllProblemTags() {
-        // Get all problem tags and convert them to DTOs.
+    public List<ProblemTagDto> getAllProblemTags(String token) {
+        // Get all problem tags this person owns and convert them to DTOs.
+        String uid = service.verifyToken(token);
+
         List<ProblemTagDto> problemTagDtos = new ArrayList<>();
-        problemTagRepository.findAll().forEach(problemTag -> problemTagDtos.add(ProblemMapper.toProblemTagDto(problemTag)));
+        problemTagRepository.findAllByOwner_Uid(uid).forEach(problemTag -> problemTagDtos.add(ProblemMapper.toProblemTagDto(problemTag)));
         return problemTagDtos;
     }
```

Comment:
I think this all makes sense - about the duplicate name issue thing, I fixed this issue by editing the code/database to remove that unique constraint as you mentioned, so now even if a tag exists in someone else's account, they'll still be able to create it. At the same time, they'll still be barred from creating a tag if another one with the same name exists among their own tags, so users won't be able to have duplicate tag names on a problem. 

I agree about ideally having verified tags in the future, similar to how LeetCode does it for example (that way we can have a set of standard tags that everyone can use privately) - I'll probably leave this for a future PR tho as you mentioned. Of those three ideas, I think 3 makes the most sense as well, since that way we can quality control "verified" tags and don't have to deal with the many issues of anyone creating public tags (e.g. tags that are deleted by another owner, having to manage displaying a potentially super long list of private, verified, and public tags, etc.)

Reply:
Sounds good! 👍